desktop/src/components/Terminal/useStreamingTerminal.tsx

```diff
@@ -1,47 +1,96 @@
 import dayjs from "dayjs"
 import { Theme, useToken } from "@chakra-ui/react"
-import { useCallback, useMemo, useRef } from "react"
-import { TStreamEventListenerFn } from "../../client"
-import { TLogOutput } from "../../types"
+import React, { useCallback, useMemo, useRef } from "react"
+import { TStreamEventListenerFn } from "@/client"
+import { TLogOutput } from "@/types"
 import { Terminal, TTerminal } from "./Terminal"
+import { TSearchOptions, useTerminalSearch } from "@/components/Terminal/useTerminalSearch"
 
 export function useStreamingTerminal({
   fontSize,
-}: { fontSize?: keyof Theme["fontSizes"] } | undefined = {}) {
+  searchOptions,
+}:
+  | {
+      fontSize?: keyof Theme["fontSizes"]
+      searchOptions?: TSearchOptions
+    }
+  | undefined = {}) {
   const terminalRef = useRef<TTerminal | null>(null)
+
+  const {
+    internals: { searchStateRef, debounceSearchResults, resetSearch, onResize },
+    searchApi,
+  } = useTerminalSearch(terminalRef, searchOptions)
+
   const fontSizeToken = useToken(
     "fontSizes",
     useMemo(() => fontSize ?? "md", [fontSize])
   )
+
   const terminal = useMemo(
-    () => <Terminal ref={terminalRef} fontSize={fontSizeToken} />,
-    [fontSizeToken]
+    () => <Terminal ref={terminalRef} fontSize={fontSizeToken} onResize={onResize} />,
+    [fontSizeToken, onResize]
   )
+
   const connectStream = useCallback<TStreamEventListenerFn>(
     (event) => {
       switch (event.type) {
-        case "data":
+        case "data": {
           if (event.data.message === undefined) {
             return
           }
-          terminalRef.current?.writeln(formatLine(event.data))
+
+          const formattedLine = formatLine(event.data)
+          terminalRef.current?.writeln(formattedLine.ansi)
+
+          searchStateRef.current.preWrappedLines = undefined
+          searchStateRef.current.searchableLines.push(...processInputLine(formattedLine.plain))
```

Comment:
There isn't a good way to reuse the terminals internal state I suppose? 
Is it because of formatting and line breaks/wraps?

Reply:
It's not a very fun API to work with. At this stage you will get the wrong length for the underlying internal buffer (the actual write is delayed via setTimeout internally...) and cannot really access the data properly. I wouldn't rule it out that there is a way to somehow get notified when the write occurs, but at this point I found it much saner to just wrap the lines ourselves rather than trying to build around the terminal's way of doing things.

If we keep running into desync issues (search results appear where they shouldn't) we can revisit this, but right now it works on all the cases that I'm able to test.